src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -62,6 +62,21 @@ _mongoc_cursor_set_opt_int64 (mongoc_cursor_t *cursor,
 }
 
 
+static const bson_value_t *
+_mongoc_cursor_get_opt (const mongoc_cursor_t *cursor, const char *option)
+{
+   bson_iter_t iter;
+
+   if (bson_iter_init_find (&iter, &cursor->opts, option)) {
+      /* TODO: bson_iter_value's return value is invalidated if the iter is
+       * modified. We may need to copy into a bson_value_t* dst instead. */
+      return bson_iter_value (&iter);
```

Comment:
This is an outstanding question. The tests seem OK as-is, but I'm curious if we should address this.

Does `bson_iter_value` actually depend on the stack-allocated `bson_iter_t iter` or is it really depending on `cursor->opts`?

Reply:
Function is now inlined. I debated adding a comment to consider adding `_mongoc_cursor_get_opt` down the line if needed, but there's no guarantee we'd remember that comment should another case come up.